### DISCUSSION TITLE
Kidnap:  Affect odom?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#485] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Mac OS, Fusion VM, Ubuntu 20.04

### Which version of ROS 2 is installed on your computer?

Galactic

### Which firmware version is installed on your robot?

G.5.3

### Which RMW is your robot running?

CycloneDDS

### Is the robot connected to a network? If so, what is the network type?

Yes: 2.4GHz Home WiFi 

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

Yes

### What is the Adapter Board's USB/BLE Toggle currently switched to?

Bluetooth (default)

### Describe your question.

Does kidnap affect odometry? 

Is orientation estimate updated using the imu? 

There are non-zero x,y pose estimates in the odometry post kidnap.  What do they reference?

If the bot is kidnapped then set down, is there a way to know the bot was kidnapped (if the node was not subscribed to /kidnap_status)?

**Comment by alsora** [comment, marked as answer]
> Does kidnap affect odometry?

Yes, kidnapping the robot will affect the odometry in a negative way.
The Imu will be updating the orientation estimate while you suddenly move the robot. 
It's also important to remember that the odometry provided by the robot is computed only through deadreckoning.
The pose estimate will drift over time. It's mostly suitable for local motion control.
You should augment the system with a slam algorithm to track the robot in a global frame.

> Is orientation estimate updated using the imu?

Yes

> There are non-zero x,y pose estimates in the odometry post kidnap. What do they reference?

Kidnapping affects the odometry. You shouldn't rely on the shown odometry value. I would recommend to reset the odometry after a kidnap, using the dedicated ROS2 service

> If the bot is kidnapped then set down, is there a way to know the bot was kidnapped (if the node was not subscribed to /kidnap_status)?

You should subscribe to that topic if you want to know when the robot is kidnapped.
